cmd/clusterctl/client/config_test.go

```diff
@@ -476,27 +465,18 @@ func Test_clusterctlClient_GetClusterTemplate(t *testing.T) {
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
 			got, err := client.GetClusterTemplate(tt.args.options)
-			if tt.wantErr != (err != nil) {
-				t.Fatalf("error = %v, wantErr = %v", err, tt.wantErr)
-			}
 			if tt.wantErr {
+				g.Expect(err).To(HaveOccurred())
 				return
 			}
 
-			if !reflect.DeepEqual(got.Variables(), tt.want.variables) {
-				t.Errorf("Variables() got = %v, want %v", got.Variables(), tt.want.variables)
-			}
-			if got.TargetNamespace() != tt.want.targetNamespace {
-				t.Errorf("TargetNamespace() got = %v, want %v", got.TargetNamespace(), tt.want.targetNamespace)
-			}
+			g.Expect(err).NotTo(HaveOccurred())
+			g.Expect(got.Variables()).To(Equal(tt.want.variables))
+			g.Expect(got.TargetNamespace()).To(Equal(tt.want.targetNamespace))
 
 			gotYaml, err := got.Yaml()
-			if err != nil {
-				t.Fatalf("Yaml() error = %v, wantErr nil", err)
-			}
-			if !reflect.DeepEqual(gotYaml, tt.want.yaml) {
-				t.Errorf("Yaml() got = %v, want %v", gotYaml, tt.want.yaml)
-			}
+			g.Expect(err).NotTo(HaveOccurred())
+			g.Expect(gotYaml).To(MatchYAML(tt.want.yaml))
```

Comment:
being tt.want.yaml an []byte like gotYaml, we should use `To(Equal(tt.want.yaml ))` thus avoiding conversion to YAML and back to string/[]bytes

Reply:
ok